### PR TITLE
chore: check out if nix-fast-build would work for us

### PR DESCRIPTION
**Pull request by dpc**

### CODE REVIEW

**Comment by elsirion** [comment]
Did we notice that nix evaluation was a significant part of CI runtime?

**Comment by dpc** [comment]
Seems like other than much uglier syntax, and not getting code coverage unless code actually changed, everything else should be supported.

File: .github/workflows/ci-nix.yml [hunk 1]
```diff
@@ -241,10 +241,12 @@ jobs:
...
       - name: Build and run tests with Code Coverage
-        run: nix build -L .#ci.workspaceCov
+        run: nix run "github:Mic92/nix-fast-build?rev=4376b8a33b217ee2f78ba3dcff01a3e464d13a46" -- --skip-cached --no-nom --flake .#legacyPackages.$(nix eval --raw --impure --expr builtins.currentSystem).ci.workspaceCov
 
-      - name: Ensure lcov.info exists
-        run: test -f result/lcov.info
+      # nix-fast-build will not create `result-` (- on purpuse) if the result was already build
+      # which is kind of OK - no code changes requires no codecov message.
+      # - name: Ensure lcov.info exists
+      #   run: test -f result-/lcov.info
```
Comment (elsirion) [inline .github/workflows/ci-nix.yml:249]: There was a reason we introduced this once, but I guess we should notice if we don't see coverage reports where there should be some …

**Review by elsirion — APPROVED** [review]
Gotta say, I'm a bit scared of fiat CI :grimacing:

**Comment by dpc** [comment]
>  Gotta say, I'm a bit scared of fiat CI 😬


https://github.com/fedimint/fedimint/pull/4272 let's test it @elsirion

**Comment by dpc** [comment]
I accidentally force pushed. Reverted to previous version, but need a rack.

**Review by bradleystachurski — DISMISSED** [review]

**Review by elsirion — DISMISSED** [review]

**Review by maan2003 — APPROVED** [review]

**Review by bradleystachurski — APPROVED** [review]

**Pull request merged**